### PR TITLE
Removing the content-type from headers as suggested by MrJoe

### PR DESCRIPTION
**Pull request by jstolp**
Removing the content-type from headers.

Untested!

### CODE REVIEW

**Comment by jstolp** [comment]
@steffjenl could you resolve the conflicts? and create a new release

**Comment by jstolp** [comment]
AS per the HTTP/1.1 RFC  (https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html)

## 14.17 Content-Type

The Content-Type entity-header field indicates the media type of the entity-body sent to the recipient or, in the case of the HEAD method, the media type that would have been sent had the request been a GET.

       Content-Type   = "Content-Type" ":" media-type
Media types are defined in section 3.7. An example of the field is

       Content-Type: text/html; charset=ISO-8859-4
Further discussion of methods for identifying the media type of an entity is provided in section 7.2.1.

## 14.13 Content-Length
The Content-Length entity-header field indicates the size of the entity-body, in decimal number of OCTETs, sent to the recipient or, in the case of the HEAD method, the size of the entity-body that would have been sent had the request been a GET.

       Content-Length    = "Content-Length" ":" 1*DIGIT
An example is

       Content-Length: 3495
Applications SHOULD use this field to indicate the transfer-length of the message-body, unless this is prohibited by the rules in section 4.4.

Any Content-Length greater than or equal to zero is a valid value. Section 4.4 describes how to determine the length of a message-body if a Content-Length is not given.

Note that the meaning of this field is significantly different from the corresponding definition in MIME, where it is an optional field used within the "message/external-body" content-type. In HTTP, it SHOULD be sent whenever the message's length can be determined prior to being transferred, unless this is prohibited by the rules in section 4.4.

File: src/SignhostClient.php [hunk 1]
```diff
@@ -47,7 +47,6 @@ public function __construct(
         array $requestOptions = []
     ) {
         $this->headers = [
-            'Content-Type: application/json',
```
Comment (talkinnl) [inline src/SignhostClient.php:50]: headers[0] is replaced in later code ('$headers[0] = 'Content-Type: application/pdf';'), so this is a bug.

That code will now replace the 'Application' header.

Solution: Replace numeric keys by the header name in all $headers array and no longer hardcode their magic order.

File: src/SignhostClient.php [hunk 2]
```diff
@@ -85,12 +84,6 @@ public function performRequest(string $endpoint, string $method, $data = null, $
                 $headers[]  = 'Digest: SHA256=' . base64_encode(pack('H*', hash_file('sha256', $filePath)));
             }
```
Comment (jstolp) [inline src/SignhostClient.php:85]: Thought I removed that. Must have missed it. Thnx.

**Pull request closed without merging**